StRoot/StFwdTrackMaker/StFwdQAMaker.cxx

```diff
@@ -0,0 +1,333 @@
+#include "StBFChain/StBFChain.h"
+#include "StFwdTrackMaker/StFwdTrackMaker.h"
+
+#include "StFwdTrackMaker/include/Tracker/FwdTracker.h"
+#include "StFwdTrackMaker/include/Tracker/ObjExporter.h"
+// StEvent includes
+#include "StEvent/StBTofCollection.h"
+#include "StEvent/StBTofHeader.h"
+#include "StEvent/StEvent.h"
+#include "StEvent/StFttCluster.h"
+#include "StEvent/StFttCollection.h"
+#include "StEvent/StFcsCluster.h"
+#include "StEvent/StFcsCollection.h"
+#include "StFcsDbMaker/StFcsDb.h"
+#include "StRoot/StEpdUtil/StEpdGeom.h"
+#include "StEvent/StFwdTrackCollection.h"
+#include "StEvent/StFwdTrack.h"
+
+
+#include "StMuDSTMaker/COMMON/StMuDstMaker.h"
+#include "StMuDSTMaker/COMMON/StMuDst.h"
+#include "StMuDSTMaker/COMMON/StMuEvent.h"
+#include "StMuDSTMaker/COMMON/StMuFstCollection.h"
+#include "StMuDSTMaker/COMMON/StMuFstHit.h"
+#include "StMuDSTMaker/COMMON/StMuPrimaryVertex.h"
+#include "StMuDSTMaker/COMMON/StMuFwdTrack.h"
+#include "StMuDSTMaker/COMMON/StMuFwdTrackCollection.h"
+#include "StMuDSTMaker/COMMON/StMuFcsCollection.h"
+#include "StMuDSTMaker/COMMON/StMuFcsCluster.h"
+#include "StMuDSTMaker/COMMON/StMuFcsHit.h"
+#include "StMuDSTMaker/COMMON/StMuFttCluster.h"
+#include "StMuDSTMaker/COMMON/StMuFttPoint.h"
+#include "StMuDSTMaker/COMMON/StMuMcTrack.h"
+#include "StMuDSTMaker/COMMON/StMuFstHit.h"
+
+// ClassImp(FcsClusterWithStarXYZ);
+
+/** Clear the FwdQATreeData from one event to next */
+void FwdQATreeData::clear(){
+    header.clear();
+    mcTracks.reset();
+    fttPoints.reset();
+    fttClusters.reset();
+    fstPoints.reset();
+    reco.reset();
+    seeds.reset();
+    wcal.reset();
+    hcal.reset();
+    wcalHits.reset();
+    hcalHits.reset();
+    epdHits.reset();
+}
+
+FcsClusterWithStarXYZ::FcsClusterWithStarXYZ( StMuFcsCluster *clu, StFcsDb *fcsDb ) {
+    if ( nullptr == clu ) return;
+    StThreeVectorD xyz = fcsDb->getStarXYZfromColumnRow(clu->detectorId(),clu->x(),clu->y());
+    float detOffset = 0.0;
+    if ( clu->detectorId() == kFcsEcalNorthDetId || clu->detectorId() == kFcsEcalSouthDetId ){
+        detOffset = 715.0; // cm from IP
+    } else if ( clu->detectorId() == kFcsHcalNorthDetId || clu->detectorId() == kFcsHcalSouthDetId ){
+        detOffset = 807.0; // cm from IP
+    }
+    mXYZ.SetXYZ( xyz.x(), xyz.y(), xyz.z() + detOffset );
+    mClu = clu;
+}
+
+FcsHitWithStarXYZ::FcsHitWithStarXYZ( StMuFcsHit *hit, StFcsDb *fcsDb ) {
+    if ( nullptr == hit ) return;
+    StThreeVectorD xyz = fcsDb->getStarXYZ(hit->detectorId(),hit->id());
+    float detOffset = 0.0;
+    if ( hit->detectorId() == kFcsEcalNorthDetId || hit->detectorId() == kFcsEcalSouthDetId ){
+        detOffset = 715.0; // cm from IP
+    } else if ( hit->detectorId() == kFcsHcalNorthDetId || hit->detectorId() == kFcsHcalSouthDetId ){
+        detOffset = 807.0; // cm from IP
+    } else if ( hit->detectorId() == kFcsPresNorthDetId || hit->detectorId() == kFcsPresSouthDetId ){
+        StEpdGeom epdgeo;
```

Comment:
Maybe something I don't know about: why the EPD Geometry for `FcsPres` detectors?